src/Microsoft.Sbom.Parsers.Spdx22SbomParser/Utils/InternalMetadataProviderIdentityExtensions.cs

```diff
@@ -168,6 +168,25 @@ public static string GetDocumentNamespace(this IInternalMetadataProvider interna
         return internalMetadataProvider.GetSBOMNamespaceUri();
     }
 
+    public static string GetSwidTagId(this IInternalMetadataProvider internalMetadataProvider)
+    {
+        if (internalMetadataProvider is null)
+        {
+            throw new ArgumentNullException(nameof(internalMetadataProvider));
+        }
+
+        string rootPackageVersion = internalMetadataProvider.GetPackageVersion();
+        string packageSupplierFromMetadata = internalMetadataProvider.GetPackageSupplier();
+        string rootPackageName = internalMetadataProvider.GetPackageName();
+
+        Uri namespaceUri = new Uri(internalMetadataProvider.GetSBOMNamespaceUri());
+
+        // Generate a guid for the new swid tag Id.
+        string tagId = Guid.NewGuid().ToString();
+
+        return $"pkg:swid/{packageSupplierFromMetadata}/{namespaceUri.Host}/{rootPackageName}@{rootPackageVersion}?tag_id={tagId}";
```

Comment:
Is this supposed to be a URI? If so, can we do a new URI().ToString() to validate it is really a URI?

Reply:
Made the change to use a URI but left the method to still return a string as this is what the GetSBOMNamespaceUri() method does as well.